### ISSUE TITLE
some generation problems after updating to latest support library

### ISSUE THREAD

**Issue by dbachelder** [johncarl81/transfuse#178]
running into this after updating to latest 2.2.0 support library

A DrawerLayout is being `@Inject`ed .. so transfuse is being a bit overly aggressive or something here. 

additionally the error does not specify an actual location for this error (I know where it is... just sayin')

```
error: cannot access State
  class file for android.support.v4.widget.DrawerLayout$State not found
  Consult the following stack trace for details.
  com.sun.tools.javac.code.Symbol$CompletionFailure: class file for android.support.v4.widget.DrawerLayout$State not found
```

**Comment by dbachelder** [comment]
similar and more difficult to workaround issue with DialogFragments:

```
Error:cannot access DialogStyle
  class file for android.support.v4.app.DialogFragment$DialogStyle not found
  Consult the following stack trace for details.
  com.sun.tools.javac.code.Symbol$CompletionFailure: class file for android.support.v4.app.DialogFragment$DialogStyle not found
```

**Comment by johncarl81** [comment]
Dumb question: are you using the latest build d7cbee97d5759df7747b12728c291b2b2ac41df9 ?

**Comment by dbachelder** [comment]
I am using the latest commit. Yes. Via jitpack

**Comment by johncarl81** [comment]
From what I was tinkering with this call made Transfuse much less eager: https://github.com/johncarl81/transfuse/blob/master/transfuse-support/src/main/java/org/androidtransfuse/adapter/element/ASTElementFactory.java#L81

Do you have more to the stack trace, like a specific line that is throwing the exception?

And it goes without saying: I need to go on a crusade against these sorts of RuntimeExceptions.  You definitely need better errors than Transfuse blowing up.

**Comment by dbachelder** [comment]
This is the only stack i get with --info --stacktrace:

```
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':rentals:compileStageDebugJava'.
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
	at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
	at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
	at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
	at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:42)
	at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
	at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
	at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
	at org.gradle.api.internal.AbstractTask.executeWithoutThrowingTaskFailure(AbstractTask.java:310)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.executeTask(AbstractTaskPlanExecutor.java:79)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:63)
	at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:51)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:54)
	at org.gradle.internal.concurrent.StoppableExecutorImpl$1.run(StoppableExecutorImpl.java:40)
Caused by: org.gradle.api.internal.tasks.compile.CompilationFailedException: Compilation failed; see the compiler error output for details.
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:47)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:33)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.delegateAndHandleErrors(NormalizingJavaCompiler.java:101)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:50)
	at org.gradle.api.internal.tasks.compile.NormalizingJavaCompiler.execute(NormalizingJavaCompiler.java:36)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:34)
	at org.gradle.api.internal.tasks.compile.CleaningJavaCompilerSupport.execute(CleaningJavaCompilerSupport.java:25)
	at org.gradle.api.tasks.compile.JavaCompile.performCompilation(JavaCompile.java:157)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:137)
	at org.gradle.api.tasks.compile.JavaCompile.compile(JavaCompile.java:91)
	at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.doExecute(AnnotationProcessingTaskFactory.java:243)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:219)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$IncrementalTaskAction.execute(AnnotationProcessingTaskFactory.java:230)
	at org.gradle.api.internal.project.taskfactory.AnnotationProcessingTaskFactory$StandardTaskAction.execute(AnnotationProcessingTaskFactory.java:208)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
	at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
```

**Comment by johncarl81** [comment]
This error doesn't seem to be local to Transfuse.  Both AndroidAnnotations (https://github.com/excilys/androidannotations/issues/1435) and Dagger (https://github.com/google/dagger/issues/197) are reporting the same issue.  It seems the support library is missing the `DialogStyle` class:

https://code.google.com/p/android/issues/detail?id=175086

There are a few workarounds currently.  You could either introduce a dummy class for the DialogFragment$DialogStyle` class or rebuild the appcompat pieces:

http://stackoverflow.com/questions/30526791/cannot-access-dialogstyle#answer-30539456

Let me know if this works for you.

**Comment by johncarl81** [comment]
And to your original question, I think you could solve the `DrawerLayout$State` error in the same way... either add a dummy file for that class or rebuild the appcompat pieces.

**Comment by dbachelder** [comment]
I seem to be getting NPEs with this solution at build time... 

```
java.lang.NullPointerException
	at com.sun.tools.javac.code.Types.eraseNotNeeded(Types.java:2171)
	at com.sun.tools.javac.code.Types.erasure(Types.java:2164)
	at com.sun.tools.javac.comp.TransTypes.erasure(TransTypes.java:867)
	at com.sun.tools.javac.comp.TransTypes.translateClass(TransTypes.java:1018)
	at com.sun.tools.javac.comp.TransTypes.visitClassDef(TransTypes.java:511)
	at com.sun.tools.javac.tree.JCTree$JCClassDecl.accept(JCTree.java:693)
	at com.sun.tools.javac.tree.TreeTranslator.translate(TreeTranslator.java:58)
	at com.sun.tools.javac.comp.TransTypes.translate(TransTypes.java:490)
	at com.sun.tools.javac.comp.TransTypes.translateTopLevelClass(TransTypes.java:1035)
	at com.sun.tools.javac.main.JavaCompiler.desugar(JavaCompiler.java:1486)
	at com.sun.tools.javac.main.JavaCompiler.desugar(JavaCompiler.java:1356)
	at com.sun.tools.javac.main.JavaCompiler.compile2(JavaCompiler.java:901)
	at com.sun.tools.javac.main.JavaCompiler.compile(JavaCompiler.java:860)
	at com.sun.tools.javac.main.Main.compile(Main.java:523)
	at com.sun.tools.javac.api.JavacTaskImpl.doCall(JavacTaskImpl.java:129)
	at com.sun.tools.javac.api.JavacTaskImpl.call(JavacTaskImpl.java:138)
	at org.gradle.api.internal.tasks.compile.JdkJavaCompiler.execute(JdkJavaCompiler.java:45)
```

I'll look into this further after my meeting

**Comment by johncarl81** [comment]
The SO solution mentions some people experience an NPE.  In that case you can rename the file to something else (TempFragmentDialog.java). (see second solution)

**Comment by dbachelder** [comment]
Oh man... ugly.. I'm just going to revert.. what a pain! but certainly not transfuse's fault. Thanks!

**Comment by dbachelder** [comment]
This turned into a rats nest of issues.... what do you think about taking the approach that android annotations used to get around this?

https://github.com/excilys/androidannotations/commit/08e5646eba30d5ae388ebb835eea1bd694adb01a

**Comment by johncarl81** [comment]
I was just looking at that... we can certainly do this.  We'd just need to blacklist and stub out certain classes that wouldn't need to be analyzed anyways.

**Comment by johncarl81** [comment]
@dbachelder, would you (or @justincpollard) like to contribute this update?  I think you'd just need to introduce a `ASTStringType` when android classes are detected in the `ASTElementFactory` here:  https://github.com/johncarl81/transfuse/blob/master/transfuse-support/src/main/java/org/androidtransfuse/adapter/element/ASTElementFactory.java#L97 instead of the `ASTElementType` returned in the default case.  It may not hurt to do the same for the [`ASTClassFactory`](https://github.com/johncarl81/transfuse/blob/master/transfuse-support/src/main/java/org/androidtransfuse/adapter/classes/ASTClassFactory.java)

**Comment by johncarl81** [comment]
FYI, looks like this was fixed in 2.22.1: https://code.google.com/p/android/issues/detail?id=175086#c42
